src/tools/compiletest/src/load_cfg.rs

```diff
@@ -0,0 +1,169 @@
+#![allow(unused)]
+
+mod itemlist;
+mod prepare;
+
+use error::{Error, ErrorExt, Result};
+
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+pub struct LineCol {
+    pub line: usize,
+    pub col: usize,
+}
+
+impl LineCol {
+    const fn new(line: usize, col: usize) -> Self {
+        Self { line, col }
+    }
+}
+
+/// When we need to e.g. build regexes that include a pattern, we need to know what kind of
+/// comments to use. Usually we just build a regex for all expressions, even though we don't
+/// use them.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum CommentTy {
+    Slashes,
+    Hash,
+    Semi,
+}
+
+impl CommentTy {
+    const fn as_str(self) -> &'static str {
+        match self {
+            CommentTy::Slashes => "//",
+            CommentTy::Hash => "#",
+            CommentTy::Semi => ";",
+        }
+    }
+
+    const fn directive(self) -> &'static str {
+        match self {
+            CommentTy::Slashes => "//@",
+            CommentTy::Hash => "#@",
+            CommentTy::Semi => ";@",
```

Comment:
Question: I don't think I've ported any `; <directive-names>` directives over either, are these for like assembly tests?

Reply:
I was thinking LLVM of IR, but duh that's not our job to test 🤦